pandas/tests/io/test_sql.py

```diff
@@ -1619,43 +1598,29 @@ class _TestSQLAlchemy(SQLAlchemyMixIn, PandasSQLTest):
     flavor: str
 
     @classmethod
-    @pytest.fixture(autouse=True, scope="class")
     def setup_class(cls):
-        cls.setup_import()
         cls.setup_driver()
-        conn = cls.conn = cls.connect()
-        conn.connect()
-
-    def load_test_data_and_sql(self):
-        pass
+        cls.setup_engine()
 
     @pytest.fixture(autouse=True)
-    def setup_method(self, load_iris_data, load_types_data):
-        pass
-
-    @classmethod
-    def setup_import(cls):
-        # Skip this test if SQLAlchemy not available
-        if not SQLALCHEMY_INSTALLED:
-            pytest.skip("SQLAlchemy not installed")
+    def setup_method(self, iris_path, types_data):
+        try:
+            self.conn = self.engine.connect()
+            self.conn.begin()
```

Comment:
Is the `commit` or `rollback` handled in `sql.SQLDatabase`?

Reply:
`sql.SQLDatabase` doesn't commit or rollback, but `sql.SQLiteDatabase` does both.  I clarified this behavior in the `to_sql` documentation in #49531 after confirming that it's true in pandas version 1.5.1.

Reply:
Okay great thanks for confirming